src/shared/utilities/tsUtils.ts

```diff
@@ -145,3 +145,5 @@ export type Mutable<T> = { -readonly [P in keyof T]: T[P] }
 export type FactoryFunction<T extends abstract new (...args: any[]) => any> = (
     ...args: ConstructorParameters<T>
 ) => InstanceType<T>
+
+export type NumericKeys<T> = { [P in keyof T]-?: T[P] extends number | undefined ? P : never }[keyof T]
```

Comment:
is there a brief docstring that can give an intuitive description of the practical purpose of this